File: src/MultiWall_cs/MultiWall/BoundsUtilities.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;

namespace MultiWall
{
    class BoundsUtilities
    {
        public static Rectangle AddBounds(Rectangle sourceBounds, Rectangle newBounds)
        {
            if (newBounds.Right > sourceBounds.Right)
                sourceBounds.Width += (newBounds.Right - sourceBounds.Width);

            if (newBounds.Bottom > sourceBounds.Bottom)
                sourceBounds.Height += (newBounds.Bottom - sourceBounds.Height);

            if (newBounds.Left < sourceBounds.Left)
            {
                sourceBounds.X = newBounds.X;
            }

            if (newBounds.Top < sourceBounds.Top)
            {
                sourceBounds.Y = newBounds.Y;
            }

            return sourceBounds;
        }
    }
}

File: src/MultiWall_vb/MultiWall/BoundsUtilities.vb
Public Class BoundsUtilities
    Public Shared Function AddBounds(ByVal sourceBounds As Rectangle, ByVal newBounds As Rectangle) As Rectangle
        If newBounds.Right > sourceBounds.Right Then
            sourceBounds.Width += (newBounds.Right - sourceBounds.Width)
        End If

        If newBounds.Bottom > sourceBounds.Bottom Then
            sourceBounds.Height += (newBounds.Bottom - sourceBounds.Height)
        End If

        If newBounds.Left < sourceBounds.Left Then
            sourceBounds.X = newBounds.X
        End If

        If newBounds.Top < sourceBounds.Top Then
            sourceBounds.Y = newBounds.Y
        End If

        Return sourceBounds
    End Function
End Class
